ExporterDirectorDistributionTest.shouldNotResetExporterPositionWhenOldPositionReceived is flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? 
- Does it block your work?
- Do we suspect that it is a real failure? No

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
-------------------------------------------------------------------------------
Test set: io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest
-------------------------------------------------------------------------------
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 15.11 s <<< FAILURE! - in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest
io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldNotResetExporterPositionWhenOldPositionReceived  Time elapsed: 14.456 s  <<< ERROR!
org.awaitility.core.ConditionTimeoutException: 
Condition with alias 'Active Director has distributed positions and passive has received it' didn't complete within 10 seconds because assertion condition defined as a lambda expression in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest that uses io.camunda.zeebe.broker.exporter.stream.ExportersState 
expected: 10L
 but was: -1L.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldNotResetExporterPositionWhenOldPositionReceived(ExporterDirectorDistributionTest.java:123)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: org.opentest4j.AssertionFailedError: 
expected: 10L
 but was: -1L
	at jdk.internal.reflect.GeneratedConstructorAccessor31.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.lambda$shouldNotResetExporterPositionWhenOldPositionReceived$5(ExporterDirectorDistributionTest.java:125)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)


</pre>
</details>

**Hypotheses**

It appears that we set the position before the exporter has recovered from the snapshot.

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
11:30:43.680 [Broker-0-LogStream-1] [-zb-actors-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
11:30:43.811 [Broker-0-LogStream-1] [-zb-actors-0] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
11:30:45.739 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
11:30:45.844 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
11:30:46.114 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
11:30:46.113 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
11:30:46.121 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
11:30:46.121 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
11:30:46.126 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
11:30:46.126 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
11:30:46.141 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND=true, COMMAND_REJECTION=true, NULL_VAL=true, SBE_UNKNOWN=true, EVENT=true}, acceptValueTypes={NULL_VAL=true, VARIABLE=true, MESSAGE_SUBSCRIPTION=true, JOB=true, TIMER=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, DEPLOYMENT_DISTRIBUTION=true, PROCESS_INSTANCE_RESULT=true, INCIDENT=true, SBE_UNKNOWN=true, DEPLOYMENT=true, MESSAGE=true, PROCESS_INSTANCE=true, JOB_BATCH=true, ERROR=true, PROCESS_EVENT=true, PROCESS=true, PROCESS_INSTANCE_CREATION=true, PROCESS_MESSAGE_SUBSCRIPTION=true, VARIABLE_DOCUMENT=true}}
11:30:46.140 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND=true, COMMAND_REJECTION=true, NULL_VAL=true, SBE_UNKNOWN=true, EVENT=true}, acceptValueTypes={NULL_VAL=true, VARIABLE=true, MESSAGE_SUBSCRIPTION=true, JOB=true, TIMER=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, DEPLOYMENT_DISTRIBUTION=true, PROCESS_INSTANCE_RESULT=true, INCIDENT=true, SBE_UNKNOWN=true, DEPLOYMENT=true, MESSAGE=true, PROCESS_INSTANCE=true, JOB_BATCH=true, ERROR=true, PROCESS_EVENT=true, PROCESS=true, PROCESS_INSTANCE_CREATION=true, PROCESS_MESSAGE_SUBSCRIPTION=true, VARIABLE_DOCUMENT=true}}
11:30:46.159 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-1'
11:30:46.161 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-2'
11:30:46.170 [exporter] [-zb-fs-workers-0] ERROR io.camunda.zeebe.broker.exporter - Actor 'exporter' failed in phase STARTED with: java.lang.IllegalStateException: Expected to find event with the snapshot position 10 in log stream, but nothing was found. Failed to recover 'exporter'. .
java.lang.IllegalStateException: Expected to find event with the snapshot position 10 in log stream, but nothing was found. Failed to recover 'exporter'.
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirector.startActiveExportingMode(ExporterDirector.java:319) ~[classes/:?]
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirector.onActorStarted(ExporterDirector.java:197) ~[classes/:?]
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) [zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:95) [zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) [zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:192) [zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
11:30:56.241 [Broker-0-LogStream-1] [-zb-actors-4] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
11:30:56.254 [stream-write-buffer] [-zb-actors-3] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
11:30:56.256 [Broker-0-LogStream-1] [-zb-actors-3] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 1 readers
11:30:56.257 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
11:30:56.258 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
11:30:56.261 [] [-zb-actors-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
11:30:56.261 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
11:30:56.269 [Broker-0-LogStream-1] [-zb-actors-5] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
11:30:56.270 [stream-write-buffer] [-zb-actors-5] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
11:30:56.271 [Broker-0-LogStream-1] [-zb-actors-5] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 0 readers
11:30:56.271 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
11:30:56.271 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
11:30:56.272 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
11:30:56.273 [] [-zb-actors-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
11:30:56.452 [Broker-0-LogStream-1] [-zb-actors-5] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
11:30:56.494 [Broker-0-LogStream-1] [-zb-actors-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
11:30:56.589 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
11:30:56.597 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
11:30:56.598 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
11:30:56.598 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
11:30:56.600 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND=true, COMMAND_REJECTION=true, NULL_VAL=true, SBE_UNKNOWN=true, EVENT=true}, acceptValueTypes={NULL_VAL=true, VARIABLE=true, MESSAGE_SUBSCRIPTION=true, JOB=true, TIMER=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, DEPLOYMENT_DISTRIBUTION=true, PROCESS_INSTANCE_RESULT=true, INCIDENT=true, SBE_UNKNOWN=true, DEPLOYMENT=true, MESSAGE=true, PROCESS_INSTANCE=true, JOB_BATCH=true, ERROR=true, PROCESS_EVENT=true, PROCESS=true, PROCESS_INSTANCE_CREATION=true, PROCESS_MESSAGE_SUBSCRIPTION=true, VARIABLE_DOCUMENT=true}}
11:30:56.604 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-1'
11:30:56.605 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-2'
11:30:56.662 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
11:30:56.670 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
11:30:56.670 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
11:30:56.671 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
11:30:56.673 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND=true, COMMAND_REJECTION=true, NULL_VAL=true, SBE_UNKNOWN=true, EVENT=true}, acceptValueTypes={NULL_VAL=true, VARIABLE=true, MESSAGE_SUBSCRIPTION=true, JOB=true, TIMER=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, DEPLOYMENT_DISTRIBUTION=true, PROCESS_INSTANCE_RESULT=true, INCIDENT=true, SBE_UNKNOWN=true, DEPLOYMENT=true, MESSAGE=true, PROCESS_INSTANCE=true, JOB_BATCH=true, ERROR=true, PROCESS_EVENT=true, PROCESS=true, PROCESS_INSTANCE_CREATION=true, PROCESS_MESSAGE_SUBSCRIPTION=true, VARIABLE_DOCUMENT=true}}
11:30:56.861 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Received new exporter state {exporter-2=1, exporter-1=1}
11:30:56.944 [Broker-0-LogStream-1] [-zb-actors-2] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
11:30:56.946 [stream-write-buffer] [-zb-actors-3] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
11:30:56.946 [Broker-0-LogStream-1] [-zb-actors-3] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 1 readers
11:30:56.947 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
11:30:56.947 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
11:30:56.947 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
11:30:56.948 [] [-zb-actors-4] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
11:30:56.953 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
11:30:56.956 [stream-write-buffer] [-zb-actors-0] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
11:30:56.957 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 0 readers
11:30:56.957 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
11:30:56.957 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
11:30:56.957 [] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
11:30:56.957 [] [-zb-actors-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

</pre>
</details>

--- a/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
+++ b/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
@@ -150,6 +150,7 @@ public final class ExporterDirectorDistributionTest {
 
     activeExporters.getClock().addTime(DISTRIBUTION_INTERVAL);
     Awaitility.await("Active Director has distributed positions and passive has received it")
+        .conditionEvaluationListener(new ClockShifter(activeExporters.getClock()))
         .untilAsserted(
             () ->
                 assertThat(passiveExporterState.getPosition(EXPORTER_ID_2))

--- a/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
+++ b/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
@@ -167,13 +167,8 @@ public final class ExporterDirectorDistributionTest {
    * <p>This makes sure that even if we miss one export position event, we distribute the event
    * later again, which makes tests less flaky.
    */
-  private static final class ClockShifter implements ConditionEvaluationListener<Void> {
-
-    private final ControlledActorClock clock;
-
-    public ClockShifter(final ControlledActorClock clock) {
-      this.clock = clock;
-    }
+  private record ClockShifter(ControlledActorClock clock)
+      implements ConditionEvaluationListener<Void> {
 
     @Override
     public void conditionEvaluated(final EvaluatedCondition<Void> condition) {

